src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,505 @@
+  /**
+  Given a series of anomaly scores, compute the likelihood for each score. This
+  function should be called once on a bunch of historical anomaly scores for an
+  initial estimate of the distribution. It should be called again every so often
+  (say every 50 records) to update the estimate. 
+
+  :param anomalyScores: a list of records. Each record is a list with the
+                        following three elements: [timestamp, value, score]
+
+                        Example::
+
+                            [datetime.datetime(2013, 8, 10, 23, 0), 6.0, 1.0]
+
+                        For best results, the list should be between 1000
+                        and 10,000 records
+  :param averagingWindow: integer number of records to average over
+  :param skipRecords: integer specifying number of records to skip when
+                      estimating distributions. If skip records are >=
+                      len(anomalyScores), a very broad distribution is returned
+                      that makes everything pretty likely.
+  :param verbosity: integer controlling extent of printouts for debugging
+
+                      0 = none
+                      1 = occasional information
+                      2 = print every record
+
+  :returns: 3-tuple consisting of:
+
+            - likelihoods
+
+              numpy array of likelihoods, one for each aggregated point
+
+            - avgRecordList
+
+              list of averaged input records
+
+            - params
+              a small JSON dict that contains the state of the estimator
+
+  **/
+  if (verbosity > 1) {
+    cout << "In estimateAnomalyLikelihoods."<<endl;
+    cout << "Number of anomaly scores:" <<  anomalyScores.size() << endl;
+    cout << "Skip records="<<  skipRecords << endl;
+//    print("First 20:", anomalyScores[0:min(20, len(anomalyScores))])
+  }
+
+  assert(anomalyScores.size() > 0); // "Must have at least one anomalyScore"
+  auto dataValues = anomalyScores; //FIXME the "data" should be anomaly scores, or raw values? 
+
+  // Estimate the distribution of anomaly scores based on aggregated records
+  if (dataValues.size()  <= skipRecords) {
+    this->distribution = nullDistribution();
+  } else {
+    dataValues.erase(dataValues.begin(), dataValues.begin() + skipRecords);// remove first skipRecords
+    this->distribution = estimateNormal(dataValues);
+  
+    /* HACK ALERT! The CLA model currently does not handle constant metric values //TODO can this be removed now? 4/2017
+     very well (time of day encoder changes sometimes lead to unstable SDR's
+     even though the metric is constant). Until this is resolved, we explicitly
+     detect and handle completely flat metric values by reporting them as not
+     anomalous.
+    */
+      auto metricDistribution = estimateNormal(dataValues, false);
```

Comment:
You're passing in the anomaly scores. You should pass in the raw values. This is detecting the case where the raw values are repeatedly the same.